Makefile

```diff
@@ -211,20 +227,22 @@ build/$(CONVERTER_JAR): $(CONVERTER_SOURCES) $(RESOURCES)
 
 build/test/cpptests: $(CPP_TEST_SOURCES) $(CPP_TEST_HEADER) $(SOURCES) $(HEADERS) $(RESOURCES) $(JAVA_HELPER_CLASSES)
 	mkdir -p build/test
+ifneq ($(RUN_CPP_TESTS),)
```

Comment:
Too much conditional execution - not in the declarative spirit of Makefiles.
In fact, you can get without conditional execution at all: the only thing you need is one conditional target. Let `test` target depend on `$(TEST_CPP)` target, which is equal to `test-cpp` if `TESTS` variable is unset or an empty string otherwise.

Reply:
E.g.:
```Makefile
ifeq ($(TESTS),)
    TEST_CPP=test-cpp
endif

test: $(TEST_CPP) test-java
```